crates/ide-assists/src/handlers/remove_unused_param.rs

```diff
@@ -96,12 +96,20 @@ fn process_usages(
 ) {
     let source_file = ctx.sema.parse(file_id);
     builder.edit_file(file_id);
-    for usage in references {
-        if let Some(text_range) = process_usage(&source_file, usage, arg_to_remove, is_self_present)
-        {
-            builder.delete(text_range);
+    let possible_ranges = references
+        .into_iter()
+        .filter_map(|usage| process_usage(&source_file, usage, arg_to_remove, is_self_present));
+
+    let mut ranges_to_delete: Vec<TextRange> = vec![];
```

Comment:
Nit:

```suggestion
    let mut ranges_to_delete: Vec<TextRange> = Vec::with_capacity(possible_ranges.len());
```

Reply:
To do this, I'd need to `collect` the `possible_ranges` iterator. I don't know enough about the performance characteristics of `Vec` to know which approach would be more efficient.

Do you think the benefit of `with_capacity` outweighs the potential cost of `collect`?

Reply:
Ah, sorry, I missed that it's an iterator. It's fine to leave it like this.